tests/ExceptionTest.php

```diff
@@ -46,6 +46,28 @@ public function testBasic(): void
         self::assertMatchesRegularExpression('~333~', $ret);
     }
 
+    public function testColorfulText(): void
+    {
+        $m = new Exception('TestIt');
+
+        $ret = $m->getColorfulText();
+        self::assertStringContainsString("\e[", $ret);
+        self::assertStringNotContainsString('\e[', $ret);
+        self::assertMatchesRegularExpression('~\e\[0m(?!.*\e\[[0-9;]*m)~s', $ret); // reset colors at the end
+
+        $m->addMoreInfo('a1', 111);
+        $ret = $m->getColorfulText();
+        self::assertStringContainsString("\e[", $ret);
+        self::assertStringNotContainsString('\e[', $ret);
+        self::assertMatchesRegularExpression('~\e\[0m(?!.*\e\[[0-9;]*m)~s', $ret); // reset colors at the end
+
+        $m->addSolution('Simple solution');
```

Comment:
add all at once, text would be then deduplicated by itself ;-)

Reply:
Not exactly. First I test simple exception, then test with added additional info and then with added solution info.

Reply:
Ahh, I know what you mean.

We can newly assert format tokens are in pairs :) - see what I meant the test is too specific..

Reply:
Sorry I don't get what you want to do differently here :)
Maybe I'm to sleepy :)